sparta/src/StatisticsArchives.cpp

```diff
@@ -196,6 +196,15 @@ void RootArchiveNode::saveTo(const std::string & dir)
     archive_controller_->saveTo(dir);
 }
 
+ArchiveNode::ArchiveNode(const std::string & name) :
+    name_(name)
+{
+}
+
+ArchiveNode::~ArchiveNode()
+{
+}
+
```

Comment:
Moving these to the `.cpp` file helps avoid some pitfalls when you use an incomplete type with a smart pointer.